src/main.py

```diff
@@ -39,128 +56,170 @@ async def on_close() -> None:
         await bc.finish(say_goodbye=False)
 
 
-@client.event
-async def on_message(message: Message) -> None:
-    global controllers
-    global loaded_image
+# Allow only one async routine to calculate list at a time
+list_task_control_lock = asyncio.Lock()
+
 
-    if message.author == client.user:
+# List command
+@client.slash_command()
```

Comment:
It would be cool to add a quick `description` to each command that's shown to users.